src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,249 @@
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.service.RetrieveCoreObjectService;
+import uk.gov.hmcts.darts.courthouse.CourthouseRepository;
+import uk.gov.hmcts.darts.dailylist.enums.JobStatusType;
+import uk.gov.hmcts.darts.dailylist.enums.SourceType;
+import uk.gov.hmcts.darts.dailylist.model.Advocate;
+import uk.gov.hmcts.darts.dailylist.model.CitizenName;
+import uk.gov.hmcts.darts.dailylist.model.CourtList;
+import uk.gov.hmcts.darts.dailylist.model.DailyList;
+import uk.gov.hmcts.darts.dailylist.model.Defendant;
+import uk.gov.hmcts.darts.dailylist.model.Hearing;
+import uk.gov.hmcts.darts.dailylist.model.Sitting;
+import uk.gov.hmcts.darts.dailylist.repository.DailyListRepository;
+import uk.gov.hmcts.darts.dailylist.service.DailyListProcessor;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DailyListProcessorImpl implements DailyListProcessor {
+    public static final String DL_TIME_NOT_BEFORE = "NOT BEFORE ";
+    public static final String DL_TIME_SITTING_AT = "SITTING AT ";
+    public static final String TIME_MARKING_NOTE_FORMAT = "hh:mm a";
+    public static final String SITTING_AT_FORMAT = "HH:mm:ss";
+
+    private final DailyListRepository dailyListRepository;
+    private final RetrieveCoreObjectService retrieveCoreObjectService;
+    private final CourthouseRepository courthouseRepository;
+    private final HearingRepository hearingRepository;
+    private final ObjectMapper objectMapper;
+
+    @Override
+    @Transactional
+    public void processAllDailyLists(LocalDate date) {
+        Arrays.stream(SourceType.values()).forEach(sourceType -> processDailyListForSourceType(date, sourceType));
+    }
+
+    private void processDailyListForSourceType(LocalDate date, SourceType sourceType) {
+        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
+        for (CourthouseEntity courthouse : allCourthouses) {
+            List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
+                    courthouse.getId(),
+                    String.valueOf(JobStatusType.NEW),
+                    date, String.valueOf(sourceType)
+            );
+
+            // Daily lists are being ordered descending by date so first item will be the most recent version
+            if (!dailyLists.isEmpty()) {
+                try {
+                    processDailyList(dailyLists.get(0));
+                } catch (JsonProcessingException e) {
+                    log.error("Failed to process dailylist for courthouse: {} with dailylist id: {}", courthouse, dailyLists.get(0).getId());
```

Comment:
This should mark the Daily List status as FAILED

Reply:
Accept. are we doing anything with failed jobs ?

Reply:
dont think so, but might need to ask @davet1985